x-pack/plugins/lens/public/types.ts

```diff
@@ -181,7 +181,10 @@ export interface Datasource<T = unknown, P = unknown> {
   getDatasourceSuggestionsFromCurrentState: (state: T) => Array<DatasourceSuggestion<T>>;
 
   getPublicAPI: (props: PublicAPIProps<T>) => DatasourcePublicAPI;
-  getErrorMessages: (state: T) => Array<{ shortMessage: string; longMessage: string }> | undefined;
+  getErrorMessages: (
+    state: T,
+    groups?: VisualizationDimensionGroupConfig[]
```

Comment:
You added the groups as an optional parameter here, but in the places `getErrorMessages` is actually called they are not passed in (in workspace and suggestion panel)